block-storage/src/main/scala/coop/rchain/blockstorage/util/io/LmdbDbi.scala

```diff
@@ -0,0 +1,65 @@
+package coop.rchain.blockstorage.util.io
+
+import java.io.{IOException, PrintWriter, StringWriter}
+
+import cats.implicits._
+import cats.effect.{ExitCase, Sync}
+import coop.rchain.blockstorage.util.io.IOError.RaiseIOError
+import coop.rchain.shared.Log
+import coop.rchain.shared.Resources.withResource
+import org.lmdbjava._
+
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+final case class LmdbDbi[F[_]: Sync: Log: RaiseIOError, T](
+    env: Env[T],
+    dbi: Dbi[T]
+) {
+  private[this] def withTxn[R](txnThunk: => Txn[T])(f: Txn[T] => R): F[R] =
+    Sync[F].bracketCase(Sync[F].delay(txnThunk)) { txn =>
```

Comment:
this will not work. you have to have the acquire-txn action in the same thread as the execution (f(txn)).
LMDB does some thread local memory allocation and will hang.

Reply:
Are you sure this is a usage bug? I have not introduced this code, I have merely refactored it into a separate class. It was originally introduced in the old [LMDBBlockStore](https://github.com/rchain/rchain/blob/dev/block-storage/src/main/scala/coop/rchain/blockstorage/LMDBBlockStore.scala#L41-L62) (judging by the commit history by you it seems) and has been working this whole time.

Reply:
true. do you need me to fix it? Im sure it will result in hard to replicate hang ups.

Reply:
I will leave this to you then since you seem to know how to fix this. Thanks!

Reply:
heres the ticket: https://rchain.atlassian.net/browse/RCHAIN-3251